pretext/project/__init__.py

```diff
@@ -85,61 +86,49 @@ class Target(pxml.BaseXmlModel, tag="target", search_mode="unordered"):
-    def post_validate(self) -> None:
-        # Select a default publication file if it's not provided.
-        if self.publication is None:
-            self.publication = Path("publication.ptx")
-            # If this publication file doesn't exist, ...
-            if not self.publication_abspath().exists():
-                # ... then use the CLI's built-in template file.
-                # TODO: this is wrong, since the returned path is only valid inside the context manager. Instead, need to enter the context here, then exit it when this class is deleted (also problematic).
-                with templates.resource_path("publication.ptx") as self.publication:
-                    pass
-        # Otherwise, verify that the provided publication file exists. TODO: perhaps skip this, instead reporting a file open error when this is read?
-        else:
-            p_full = self.publication_abspath()
-            if not p_full.exists():
-                raise FileNotFoundError(
-                    f"Provided publication file {p_full} does not exist."
-                )
-
-        # Merge in servers from the parent Project that aren't specified in this Target.
-        self_server_names = [server.name for server in self.server]
-        for server in self._project.server:
-            if server.name not in self_server_names:
-                self.server.append(server)
-
     # A path to the output directory for this target, relative to the project's `output` path.
     output_dir: Path = pxml.attr(name="output-dir", default=None)
 
-    # Make the default value for output be `self.name`. Specifying a `default_factory` won't work, since it's a `@classmethod`. So, use a validator (which has access to the object), replacing `None` (hack: a type violation) with `self.name`.
+    # Make the default value for output be `self.name`. Specifying a `default_factory` won't work, since it's a `@classmethod`. So, use a validator (which has access to the object), replacing `None` with `self.name`.
     @validator("output_dir", always=True)
-    def output_dir_defaults_to_name(cls, v: t.Optional[Path], values: t.Any) -> Path:
+    def output_dir_validator(cls, v: t.Optional[Path], values: t.Any) -> Path:
+        # When the format is Runestone, this is overwritten in `post_validate`. Make sure it's not specified.
+        if values["format"] == Format.RUNESTONE and v is not None:
+            raise ValueError("The Runestone format's output-dir must not be specified.")
         return Path(v) if v is not None else Path(values["name"])
 
-    # We validate compression before output_filename to use its value to check if we can have an output_filename
+    # We validate compression before output_filename to use its value to check if we can have an output_filename.
     compression: t.Optional[Compression] = pxml.attr()
 
+    # Compression is only supported for HTML.
```

Comment:
And webwork (i.e., webwork-sets), the build target, not the asset

Reply:
Is this correct?

``` Python
    # Compression is only supported for HTML and WeBWorK formats.
    @validator("compression")
    def compression_validator(
        cls, v: t.Optional[Compression], values: t.Any
    ) -> t.Optional[Compression]:
        if values["format"] not in (Format.HTML, Format.WEBWORK) and v is not None:
            raise ValueError("Only the HTML and WeBWorK formats support compression.")
        return v
```

Reply:
Looks good!